packages/core/src/pipes.ts

```diff
@@ -0,0 +1,195 @@
+  ef: PipeFn<E, F>,
+  fg: PipeFn<F, G>,
+): (input: A, ctx: Ctx) => G
+function resolver<A, B, C, D, E, F, G, H>(
+  ab: PipeFn<A, B>,
+  bc: PipeFn<B, C>,
+  cd: PipeFn<C, D>,
+  de: PipeFn<D, E>,
+  ef: PipeFn<E, F>,
+  fg: PipeFn<F, G>,
+  gh: PipeFn<G, H>,
+): (input: A, ctx: Ctx) => H
+function resolver<A, B, C, D, E, F, G, H, I>(
+  ab: PipeFn<A, B>,
+  bc: PipeFn<B, C>,
+  cd: PipeFn<C, D>,
+  de: PipeFn<D, E>,
+  ef: PipeFn<E, F>,
+  fg: PipeFn<F, G>,
+  gh: PipeFn<G, H>,
+  hi: PipeFn<H, I>,
+): (input: A, ctx: Ctx) => I
+function resolver<A, B, C, D, E, F, G, H, I, J>(
+  ab: PipeFn<A, B>,
+  bc: PipeFn<B, C>,
+  cd: PipeFn<C, D>,
+  de: PipeFn<D, E>,
+  ef: PipeFn<E, F>,
+  fg: PipeFn<F, G>,
+  gh: PipeFn<G, H>,
+  hi: PipeFn<H, I>,
+  ij: PipeFn<I, J>,
+): (input: A, ctx: Ctx) => J
+function resolver<A, B, C, D, E, F, G, H, I, J, K>(
+  ab: PipeFn<A, B>,
+  bc: PipeFn<B, C>,
+  cd: PipeFn<C, D>,
+  de: PipeFn<D, E>,
+  ef: PipeFn<E, F>,
+  fg: PipeFn<F, G>,
+  gh: PipeFn<G, H>,
+  hi: PipeFn<H, I>,
+  ij: PipeFn<I, J>,
+  jk: PipeFn<J, K>,
+): (input: A, ctx: Ctx) => K
+function resolver<A, B, C, D, E, F, G, H, I, J, K, L>(
+  ab: PipeFn<A, B>,
+  bc: PipeFn<B, C>,
+  cd: PipeFn<C, D>,
+  de: PipeFn<D, E>,
+  ef: PipeFn<E, F>,
+  fg: PipeFn<F, G>,
+  gh: PipeFn<G, H>,
+  hi: PipeFn<H, I>,
+  ij: PipeFn<I, J>,
+  jk: PipeFn<J, K>,
+  kl: PipeFn<K, L>,
+): (input: A, ctx: Ctx) => L
+function resolver<A, B, C, D, E, F, G, H, I, J, K, L, M>(
+  ab: PipeFn<A, B>,
+  bc: PipeFn<B, C>,
+  cd: PipeFn<C, D>,
+  de: PipeFn<D, E>,
+  ef: PipeFn<E, F>,
+  fg: PipeFn<F, G>,
+  gh: PipeFn<G, H>,
+  hi: PipeFn<H, I>,
+  ij: PipeFn<I, J>,
+  jk: PipeFn<J, K>,
+  kl: PipeFn<K, L>,
+  lm: PipeFn<L, M>,
+): (input: A, ctx: Ctx) => M
+function resolver<A, B, C, D, E, F, G, H, I, J, K, L, M, N>(
+  ab: PipeFn<A, B>,
+  bc: PipeFn<B, C>,
+  cd: PipeFn<C, D>,
+  de: PipeFn<D, E>,
+  ef: PipeFn<E, F>,
+  fg: PipeFn<F, G>,
+  gh: PipeFn<G, H>,
+  hi: PipeFn<H, I>,
+  ij: PipeFn<I, J>,
+  jk: PipeFn<J, K>,
+  kl: PipeFn<K, L>,
+  lm: PipeFn<L, M>,
+  mn: PipeFn<M, N>,
+): (input: A, ctx: Ctx) => N
+function resolver<A, B, C, D, E, F, G, H, I, J, K, L, M, N, O>(
+  ab: PipeFn<A, B>,
+  bc: PipeFn<B, C>,
+  cd: PipeFn<C, D>,
+  de: PipeFn<D, E>,
+  ef: PipeFn<E, F>,
+  fg: PipeFn<F, G>,
+  gh: PipeFn<G, H>,
+  hi: PipeFn<H, I>,
+  ij: PipeFn<I, J>,
+  jk: PipeFn<J, K>,
+  kl: PipeFn<K, L>,
+  lm: PipeFn<L, M>,
+  mn: PipeFn<M, N>,
+  no: PipeFn<N, O>,
+): (input: A, ctx: Ctx) => O
+function resolver<A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P>(
+  ab: PipeFn<A, B>,
+  bc: PipeFn<B, C>,
+  cd: PipeFn<C, D>,
+  de: PipeFn<D, E>,
+  ef: PipeFn<E, F>,
+  fg: PipeFn<F, G>,
+  gh: PipeFn<G, H>,
+  hi: PipeFn<H, I>,
+  ij: PipeFn<I, J>,
+  jk: PipeFn<J, K>,
+  kl: PipeFn<K, L>,
+  lm: PipeFn<L, M>,
+  mn: PipeFn<M, N>,
+  no: PipeFn<N, O>,
+  op: PipeFn<O, P>,
+): (input: A, ctx: Ctx) => P
+function resolver<A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q>(
```

Comment:
Just curious, why do you stop at Q?

Reply:
Because you have to stop somewhere and that's quite a lot! :)